src/Bicep.LangServer/Handlers/BicepDisableLinterRuleHandler.cs

```diff
@@ -0,0 +1,120 @@
+            File.WriteAllText(updatedBicepConfigFilePath, bicepConfigContents);
+
+            return await Unit.Task;
+        }
+
+        public (string, string) GetBicepConfigFilePathAndContents(DocumentUri documentUri, string code, string bicepConfigFilePath)
+        {
+            if (File.Exists(bicepConfigFilePath))
+            {
+                return (bicepConfigFilePath, DisableLinterRule(File.ReadAllText(bicepConfigFilePath), code));
+            }
+            else
+            {
+                var directoryContainingSourceFile = Path.GetDirectoryName(documentUri.GetFileSystemPath()) ??
+                    throw new ArgumentException("Unable to find directory information");
+
+                bicepConfigFilePath = Path.Combine(directoryContainingSourceFile, LanguageConstants.BicepConfigSettingsFileName);
+                return (bicepConfigFilePath, DisableLinterRule(string.Empty, code));
+            }
+        }
+
+        public string DisableLinterRule(string bicepConfig, string code)
+        {
+            try
+            {
+                if (JsonConvert.DeserializeObject(bicepConfig) is JObject root &&
+                    root["analyzers"] is JObject analyzers &&
+                    analyzers["core"] is JObject core)
+                {
+                    if (core["rules"] is JObject rules)
+                    {
+                        if (rules[code] is JObject ruleName)
+                        {
+                            if (ruleName.ContainsKey("level"))
+                            {
+                                ruleName["level"] = "off";
+                            }
+                            else
+                            {
+                                ruleName.Add("level", "off");
+                            }
+                        }
+                        else
+                        {
+                            SetRuleLevelToOff(rules, code);
+                        }
+                    }
+                    else
+                    {
+                        JObject rule = new JObject();
+                        SetRuleLevelToOff(rule, code);
+
+                        core.Add("rules", rule);
+                    }
+
+                    return root.ToString(Formatting.Indented);
+                }
+
+                if (JsonConvert.DeserializeObject(DefaultBicepConfig) is JObject defaultBicepConfigRoot &&
+                    defaultBicepConfigRoot["analyzers"]?["core"]?["rules"] is JObject defaultRules)
+                {
+                    SetRuleLevelToOff(defaultRules, code);
+
+                    return defaultBicepConfigRoot.ToString();
+                }
+
+                return string.Empty;
+            }
+            catch (Exception)
+            {
+                throw new Exception("File bicepconfig.json already exists and is invalid. If overwriting the file is intended, delete it manually and retry disable linter rule lightBulb option again");
```

Comment:
>Exception [](http://example.com/codeflow?start=26&length=9)

What is the user experience in VS code when this path is hit?

Reply:
Currently it shows up in the output window:
![image](https://user-images.githubusercontent.com/30270536/134120048-412cb9b7-c45d-4e30-a894-27473dabe343.png)

![image](https://user-images.githubusercontent.com/30270536/134121642-cc32fdb3-93ba-49dd-be77-a17b240fd9d2.png)

Reply:
Sounds good.